src/services/routeService.ts

```diff
@@ -56,6 +59,53 @@ export default class RouteService {
             return null;
         }
     }
+
+    /**
+     * Converts Apollo's queryResult into:
+     * @return {IExternalRoute} externalRoute
+     * @return {IExternalRoutePath[]} externalRoute.externalRoutePaths
+     * @return {IExternalRoutePathLink[]} externalRoutePaths.externalRoutePathLinks
+     * @return {IExternalRoutePathLinkNode} externalRoutePathLinks.startNode
+     * @return {IExternalRoutePathLinkNode} externalRoutePathLinks.endNode
+     */
+    private static getExternalRoute(route: any): IExternalRoute {
+
+        if (route.reitinsuuntasByReitunnus) {
+            route.externalRoutePaths = route.reitinsuuntasByReitunnus.nodes;
+            delete route.reitinsuuntasByReitunnus;
+
+            const externalRoutePaths = route.externalRoutePaths
+            .map((externalRoutePath: any) => {
+
+                externalRoutePath.externalRoutePathLinks =
+                externalRoutePath.reitinlinkkisByReitunnusAndSuuvoimastAndSuusuunta.nodes
+                .map((externalRoutePathLink: any) => {
```

Comment:
Typing as in the few before.. But is this just too much of a pain to do typings for?

Reply:
Also thinking about best practices in coding, maybe this should either create a new object and return that, or use `.forEach` rather than `.map`

Reply:
.map returns a new version of some object so that's why I used it. Not sure if forEach would fit better or not